heavy/heavy.go

```diff
@@ -96,5 +106,8 @@ func callHeavy[T any](ctx context.Context, backendURL string, cutoffBlock *uint6
 		log.Error("callHeavy response unmarshalling error", "err", err)
 		return nil, rpc.NewRPCError(-32500, genericError)
 	}
+	if IsZero[T](ret) {
```

Comment:
So, I just recently learned that you don't have to specify [T] if it can be inferred from an argument. That doesn't work for return values (unless they match an input value), but it typically would work for function parameters. That is to say, this could just be:

```
IsZero(ret)
```

And it would work just fine.